modin/utils.py

```diff
@@ -452,6 +460,30 @@ def to_pandas(modin_obj: SupportsPrivateToPandas) -> Any:
     return modin_obj._to_pandas()
 
 
+def to_numpy(
+    modin_obj: Union[SupportsPrivateToNumPy, SupportsPublicToNumPy]
+) -> np.ndarray:
+    """
+    Convert a Modin object to a NumPy array.
+
+    Parameters
+    ----------
+    modin_obj : modin.DataFrame, modin.Series, modin.numpy.array
+        The Modin distributed object to convert.
+
+    Returns
+    -------
+    numpy.array
+        Converted object with type depending on input.
+    """
+    if isinstance(modin_obj, SupportsPrivateToNumPy):
+        return modin_obj._to_numpy()
+    array = modin_obj.to_numpy()
+    if ExperimentalNumPyAPI.get():
+        array = array._to_numpy()
```

Comment:
As I understand, we can return either np.array or modin.np.array in the first line. But then we check for the experimental mode and call `_to_numpy` again, whereas `modin.np.array` doesn't have `_to_numpy` method. Seems like an issue.

Also, as I can see if `modin_obj` is a Modin DF, we always return np,array because you haven't added the things to `BasePandasDataset.to_numpy`.

Reply:
modin.numpy.array does have an internal `_to_numpy` method - which can be found here: https://github.com/modin-project/modin/pull/5422/files#diff-a5565cfb004b107c88dbae43a112a1821d007fdb560eef82c66c25d57c73cb4fR1459

I also double checked, and we do modify modin/pandas/base.py::to_numpy to return a modin.numpy.array, so even df.to_numpy() with the flag set to true will return a modin.numpy.array!